<?xml version="1.0"?>
<launch>
    <node pkg="zm_localization" type="zm_localization" name="zm_localization_node" respawn="false" output="screen">
    	<param name="broadcast_tf"              value="true"/>
        <param name="base_frame"                value="base_footprint"/>
        <param name="odom_frame"                value="odom"/>
        <param name="map_frame"                 value="map"/>
        <param name="map_size"                  value="1000"/>
        <param name="map_downscale"             value="4"/>
        <param name="map_update_rate"           value="0.5"/>
        <param name="loc_update_rate"           value="10"/>
        <param name="num_smooth"                value="2"/>
        <param name="min_score"                 value="0.01"/>
        <param name="solver_gain"               value="0.1"/>
        <param name="solver_damping"            value="1000"/>
        <param name="solver_iterations"         value="20"/>

        <param name="scan_num"                  value="2"/>
        <remap from="scan_0"                    to="/sick_s30b/laser/scan0"/>
        <remap from="scan_1"                    to="/sick_s30b/laser/scan1"/>
        <param name="sample_rate"               value="10"/>
        <param name="min_points"                value="20"/>
        <param name="update_gain"               value="0.1"/>
        <param name="confidence_gain"           value="0.1"/>
        <param name="odometry_std_xy"           value="0.01"/>
        <param name="odometry_std_yaw"          value="0.01"/>
        <param name="min_sample_std_xy"         value="0.025"/>
        <param name="min_sample_std_yaw"        value="0.025"/>
        <param name="max_sample_std_xy"         value="0.0025"/>
        <param name="max_sample_std_yaw"        value="0.0003"/>

        <param name="constrain_threshold"       value="0.0005"/>
        <param name="constrain_threshold_yaw"   value="0.0002"/>
        <param name="transform_timeout"         value="0.2"/>
    </node>
</launch>